package.json

```diff
@@ -4,24 +4,19 @@
   "description": "DXswap Decentralized Application",
   "license": "GPL-3.0-or-later",
   "scripts": {
-    "start": "yarn && react-scripts start",
-    "build": "cross-env GENERATE_SOURCEMAP=false react-app-rewired build",
-    "serve": "serve -s ./build",
-    "eject": "react-scripts eject",
+    "start": "yarn && yarn codegen:socket && react-scripts start",
+    "build": "yarn codegen:socket && cross-env GENERATE_SOURCEMAP=false react-app-rewired build",
+    "build:serve": "yarn build && serve -s ./build -l 5000",
     "ipfs-build": "cross-env PUBLIC_URL=\".\" GENERATE_SOURCEMAP=false react-app-rewired build",
     "lint": "eslint \"./src/**/*.{ts,tsx,js}\"",
-    "check-errors": "yarn lint --fix && yarn typecheck",
     "typecheck": "tsc --noEmit",
-    "test-start": "react-scripts start",
-    "cypress:ci": "cross-env start-server-and-test 'yarn test-start' http-get://localhost:3000",
-    "cypress:ct": "cross-env start-server-and-test 'yarn test-start' http-get://localhost:3000 'yarn cypress:run'",
+    "cypress:ci": "cross-env start-server-and-test 'react-scripts start' http-get://localhost:3000",
     "cypress:run": "cross-env cypress run -r mochawesome -s 'tests/cypress/integration/smoke/*'",
     "integration-test": "start-server-and-test 'serve build -l 3000' http://localhost:3000 'cypress run'",
-    "synpress:ct": "cross-env start-server-and-test 'yarn test-start' http-get://localhost:3000 'yarn synpress:run'",
+    "synpress:ct": "cross-env start-server-and-test 'react-scripts start' http-get://localhost:3000 'yarn synpress:run'",
     "synpress:open": "cross-env MOCHAWESOME_HTML='false' MOCHAWESOME_OVERWRITE='false' NETWORK_NAME='rinkeby' CYPRESS_SKIP_RESOURCES_WAIT='true' FAIL_ON_ERROR='0' PRIVATE_KEY='367f6baeb27b33408c85040e13d1ba3e2a173a9112eb3b103f79df0729ddda72' ETHERSCAN_KEY='25VF97IQZ8EFIYSYFI72UCTJJII3HSZU7Z' synpress open --configFile tests/synpress/synpress.json",
     "synpress:run": "cross-env MOCHAWESOME_HTML='false' MOCHAWESOME_OVERWRITE='false' NETWORK_NAME='rinkeby' CYPRESS_SKIP_RESOURCES_WAIT='true' FAIL_ON_ERROR='0' PRIVATE_KEY='367f6baeb27b33408c85040e13d1ba3e2a173a9112eb3b103f79df0729ddda72' ETHERSCAN_KEY='25VF97IQZ8EFIYSYFI72UCTJJII3HSZU7Z' synpress run --configFile tests/synpress/synpress.json --config supportFile='tests/support/index.js',pluginsFile='tests/plugins/index.js' -r mochawesome --reporterOptions 'overwrite=false, html=false, json=true'",
-    "codegen:socket": "TARGET=src/services/EcoBridge/Socket/api/generated && openapi-generator-cli generate --skip-validate-spec -i ./src/services/EcoBridge/Socket/api/swagger.json -g typescript-fetch -o \"$TARGET\" --additional-properties=typescriptThreePlus=true && sed -i \"1s;^;// @ts-nocheck\\n;\" $TARGET/**/*.ts",
-    "postinstall": "yarn run codegen:socket",
```

Comment:
Why should we do it outside `yarn install`?

Reply:
I have thought the same. But its good to do `yarn install && react-scripts start ` together. So if any new packages are added or changed it will be auto installed. 
So `postinstall` will run in that scenario each time. I finally came to the conclusion its ok to generate since its fast.

Reply:
If you @berteotti and @adamazad feels its an over kill. I can remove `yarn install`  from `start` as well and move this to `postinstall`

Reply:
`postinstall` is run when another project installs this repo - `@swapr/frontend`. I thought `prepare` might be it, but not. Best to remove this hook altogether.